lib/hash-password.js

```diff
@@ -4,12 +4,18 @@ var crypto = require('crypto')
 
 function hashPassword (password, salt, iterations) {
   return new Promise(function (resolve, reject) {
-    crypto.pbkdf2(password, salt, iterations, 20, 'sha1', function (err, derivedKey) {
+    var cb = function (err, derivedKey) {
       if (err) {
         reject(err)
       } else {
         resolve(derivedKey.toString('hex'))
       }
-    })
+    }
+    var version = Number(process.version.match(/^v(\d+\.\d+)/)[1])
+    if (version >= 6) {
+      crypto.pbkdf2(password, salt, iterations, 20, 'sha1', cb);
+    } else {
+      crypto.pbkdf2(password, salt, iterations, 20, cb);
+    }
```

Comment:
are you sure? I think all versions since Node 4 at least take the digest, it only used to be optional before node 8:

-  https://nodejs.org/docs/v4.8.1/api/crypto.html#crypto_crypto_pbkdf2_password_salt_iterations_keylen_digest_callback
- https://nodejs.org/docs/v5.12.0/api/crypto.html#crypto_crypto_pbkdf2_password_salt_iterations_keylen_digest_callback